### PR TITLE
Feature/civ 807 view claim screen v2

### PR DESCRIPTION
**Pull request by RosDiaf**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by taomoh — CHANGES_REQUESTED** [review]
1. I am a bit worried about the number of files added in this PR. I understand there are dependencies, but we should reduce them to the bear minimum with the nunjucks view being the anchor for the copy and only pulling the strict minimum needed with a critical view. This is what we did for the dashboard 1st thin slice (See PR: https://github.com/hmcts/civil-citizen-ui/pull/94)
Some files are not required for the View claim scope (e.g. payment code, some model and enumerations classes, etc.)

2. The following files shouldn't be modified for this story:
- bin/generate-ssl-options.sh
- bin/init.sh 

3. `package-lock.json` shouldn't be added to git. This is already covered by `yarn.lock`

4. Please also refer to inline comments I've added

File: src/main/assets/scss/main.scss [hunk 1]
```diff
@@ -1 +1,6 @@
 @import '~govuk-frontend';
+
+.bold-small {
+  font-weight: 700;
+  text-transform: none;
+}
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/main/assets/scss/main.scss]: Removed

File: src/main/views/claim-details-tpl.njk [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    ]
+  }) }}
+  <!--[if !IE 8]><!-->
+  <link href="/govuk-frontend/all.css" rel="stylesheet">
+  <!--<![endif]-->
+    {# For Internet Explorer 8, you need to compile specific stylesheet #}
+    {# see https://frontend.design-system.service.gov.uk/supporting-ie8/#support-internet-explorer-8 #}
+    <!-- [if IE 8]> <link href="/govuk-frontend/all-ie8.css" rel="stylesheet"> <![endif] -->
+    {# For older browsers to allow them to recognise HTML5 elements such as `<header>` #}
+    <!-- [if lt IE 9]> <script src="/html5-shiv/html5shiv.js"></script> <![endif] -->
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/main/views/claim-details-tpl.njk]: Code removed, not needed. I have doubled checked with Mounika.

File: package.json [hunk 3]
```diff
@@ -93,8 +109,8 @@
     "node-sass": "^6.0.1",
     "nodemon": "^2.0.15",
     "pa11y": "^6.0.1",
+    "request": "^2.88.2",
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply package.json]: Removed

File: src/main/features/response/routes/claim-details.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import { ErrorHandling } from '../../../common/errorHandling';
+// import { User } from 'app/idam/user';
+//import * as moment from 'moment'
+const moment = require('moment');
+
+// function isCurrentUserLinkedToClaim (user: User, claim: Claim): boolean {
+//   return claim.defendantId === user.id;
+// }
+
+const civilServiceApiBaseUrl = 'http://localhost:8765';
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/main/features/response/routes/claim-details.ts]: Changed to master router structure now

File: src/main/app.ts [hunk 5]
```diff
@@ -13,6 +12,9 @@ import { I18Next } from './modules/i18n';
 import { HealthCheck } from './modules/health';
 import routes from './routes/routes';
 
+import { Feature as DefendantResponseFeature } from './features/response/index';
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/main/app.ts]: Changed to master router structure

File: .gitignore [hunk 6]
```diff
@@ -25,6 +25,3 @@ dist/
 public/
 smoke-output/
 functional-output/
-
-.yarn/
-.yarnrc
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply .gitignore]: Changes reverted

File: package.json [hunk 7]
```diff
@@ -24,10 +24,14 @@
...
   "dependencies": {
+    "@hmcts/cmc-draft-store-middleware": "^2.0.1",
+    "@hmcts/cmc-validators": "^0.2.6",
+    "@hmcts/draft-store-client": "^1.2.1",
     "@hmcts/info-provider": "^1.0.0",
     "@hmcts/nodejs-healthcheck": "^1.7.0",
-    "@hmcts/nodejs-logging": "^3.0.2",
+    "@hmcts/nodejs-logging": "^3.0.4",
     "@hmcts/properties-volume": "^0.0.13",
+    "@hmcts/requestretry": "^1.1.2",
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply package.json:34]: Removed

File: src/main/app/utils/numberFormatter.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+numeral.locale('en-gb');
+
+export const NUMBER_FORMAT = '$0,0[.]00';
+
+export class NumberFormatter {
+
+  static formatMoney (value: number): string {
+    return numeral(value).format(NUMBER_FORMAT);
+  }
+}
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/main/app/utils/numberFormatter.ts]: Removed. currencyFormat used

File: package.json [hunk 9]
```diff
@@ -45,19 +50,26 @@
...
     "i18next": "^21.6.3",
     "i18next-fs-backend": "^1.1.4",
     "i18next-http-middleware": "^3.1.4",
     "jquery": "^3.5.1",
     "lodash": "^4.17.20",
+    "numeral": "^2.0.6",
     "nunjucks": "^3.2.2",
+    "nunjucks-numeral-filter": "^0.0.2",
+    "otp": "^1.0.1",
+    "request-promise-native": "^1.0.9",
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply package.json:71]: Removed

**Review by taomoh — APPROVED** [review]
Looking good. Thanks @RosDiaf for taking the time to incorporate the feedback!

**Review by mounikahmcts — APPROVED** [review]

File: src/main/app/client/civilServiceClient.ts [hunk 10]
```diff
@@ -22,4 +25,10 @@ export class CivilServiceClient {
 
     return claims;
   }
+
+  async retrieveClaimDetails(claimId: string): Promise<Claim> {
+    const response: AxiosResponse<object> = await this.client.get(`/cases/${claimId}`);
+    const objects = response.data as Claim;
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/main/app/client/civilServiceClient.ts]: I have added a try catch block

File: src/main/app/client/civilServiceClient.ts [hunk 11]
```diff
@@ -5,8 +5,11 @@ export class CivilServiceClient {
   client: AxiosInstance;
 
   constructor(baseURL: string) {
+    //const token = 'eyJ0eXAiOiJKV1QiLCJ6aXAiOiJOT05FIiwia2lkIjoiYi9PNk92VnYxK3krV2dySDVVaTlXVGlvTHQwPSIsImFsZyI6IlJTMjU2In0.eyJzdWIiOiJ0ZXN0QGV4YW1wbGUuY29tIiwiYXV0aF9sZXZlbCI6MCwiYXVkaXRUcmFja2luZ0lkIjoiZmFkOWI1YmItYTJhOC00NDViLTg0NWEtODRhOTc5OWExNTRkIiwiaXNzIjoiaHR0cDovL2ZyLWFtOjgwODAvb3BlbmFtL29hdXRoMi9obWN0cyIsInRva2VuTmFtZSI6ImFjY2Vzc190b2tlbiIsInRva2VuX3R5cGUiOiJCZWFyZXIiLCJhdXRoR3JhbnRJZCI6ImNlZGYxOTRjLWU3ZDgtNGU5YS04NzczLTYzZGVhMTIxMTczMiIsImF1ZCI6ImNjZF9nYXRld2F5IiwibmJmIjoxNjQzMTg2OTQ5LCJncmFudF90eXBlIjoiYXV0aG9yaXphdGlvbl9jb2RlIiwic2NvcGUiOlsib3BlbmlkIiwicHJvZmlsZSIsInJvbGVzIl0sImF1dGhfdGltZSI6MTY0MzE4Njk0ODAwMCwicmVhbG0iOiIvaG1jdHMiLCJleHAiOjE2NDMyMTU3NDksImlhdCI6MTY0MzE4Njk0OSwiZXhwaXJlc19pbiI6Mjg4MDAsImp0aSI6ImNhZTE2ZjIwLTFhNGYtNDlkYy04MmE4LTAzMjk2ZWQ3YzE4MiJ9.nKpuXZtq5dddvRGsvuZ24MWVuzCzfWvffIS5psHWr1ibYN7BauPN3GWWgJILZCdOn7tFEpkI12ZEtK60dFcCeQ2eqEPZHBsWNceeWinvlD9vRYEJyIjxQYpUWsDFt5jcrxXjnguBtJAlJDdacsID971iZhGOZUmm-U1BSS2mJ8enodduFz1WIT3gc6wmgd73PBuInTvRZdymHpbTJV5cG9Kf2E6BetHYlIBuysukRcVr-dkedZ0fbShgFLgzh900h6lQXGD0bjUuDJG6x7YcudxT-5kNQ8Yoo8ctEs5KbuaNvtbMMOVWXbEDjtw79ieTNvx70mag4N1Fyt6DZfq3Rg';
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/main/app/client/civilServiceClient.ts:8]: Yes, once we deploy those lines will be removed. We left there to do the demo on local machine

**Pull request merged**